habitat-baselines/habitat_baselines/config/default_structured_configs.py

```diff
@@ -0,0 +1,366 @@
+import math
+from dataclasses import dataclass, field
+from typing import Any, List, Tuple
+
+from hydra.core.config_store import ConfigStore
+from omegaconf import MISSING, II
+
+
+@dataclass
+class HabitatBaselinesBaseConfig:
+    pass
+
+
+@dataclass
+class WBConfig(HabitatBaselinesBaseConfig):
+    """Weights and Biases config"""
+    # The name of the project on W&B.
+    project_name: str = ""
+    # Logging entity (like your username or team name)
+    entity: str = ""
+    # The group ID to assign to the run. Optional to specify.
+    group: str = ""
+    # The run name to assign to the run. If not specified,
+    # W&B will randomly assign a name.
+    run_name: str = ""
+
+
+@dataclass
+class EvalConfig(HabitatBaselinesBaseConfig):
+    # The split to evaluate on
+    split: str = "val"
+    # Whether to use the config in the checkpoint. Setting this to False
+    # is useful if some code changes necessitate a new config but the weights
+    # are still valid.
+    use_ckpt_config: bool = True
+    should_load_ckpt: bool = True
+    # The number of time to run each episode through evaluation.
+    # Only works when evaluating on all episodes.
+    evals_per_ep: int = 1
+
+
+@dataclass
+class PreemptionConfig(HabitatBaselinesBaseConfig):
+    # Append the slurm job ID to the resume state filename if running
+    # a slurm job. This is useful when you want to have things from a different
+    # job but the same checkpoint dir not resume.
+    append_slurm_job_id: bool = False
+    # Number of gradient updates between saving the resume state
+    save_resume_state_interval: int = 100
+    # Save resume states only when running with slurm
+    # This is nice if you don't want debug jobs to resume
+    save_state_batch_only: bool = False
+
+
+@dataclass
+class ActionDistributionConfig(HabitatBaselinesBaseConfig):
+    use_log_std: bool = True
+    use_softplus: bool = False
+    log_std_init: float = 0.0
+    # If True, the std will be a parameter not conditioned on state
+    use_std_param: bool = False
+    # If True, the std will be clamped to the specified min and max std values
+    clamp_std: bool = True
+    min_std: float = 1e-6
+    max_std: int = 1
+    min_log_std: int = -5
+    max_log_std: int = 2
+    # For continuous action distributions (including gaussian):
+    action_activation: str = "tanh"  # ['tanh', '']
+    scheduled_std: bool = False
+
+
+@dataclass
+class ObsTransformConfig(HabitatBaselinesBaseConfig):
```

Comment:
Strange that this one is empty given that some of its implementation share common keys. Probably good enough for now, but we should try to see if there is a better way to do this better.

Reply:
In scope of this PR my goal was to define the structures from `default.py` in structured config style with minimum changes. So that we could easily compare between two.

I agree that the config inheritance of common keys has to be revisited. Maybe in subsequent PR?

Reply:
Subsequent PR it is then.